src/rpdk/core/test.py

```diff
@@ -267,3 +268,8 @@ def _sam_arguments(parser):
             "The region used for temporary credentials " f"(Default: {DEFAULT_REGION})"
         ),
     )
+    parser.add_argument(
+        "--soft-delete",
+        default="false",
```

Comment:
you can just make this a boolean so you don't have to pass in a string.

use `store_true`: https://docs.python.org/3/library/argparse.html